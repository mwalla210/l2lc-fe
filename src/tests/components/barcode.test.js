import React from 'react'
import renderer from 'react-test-renderer'
import Barcode from '../../components/barcode'
jest.mock('jsbarcode',() => {
  return jest.fn()})
import JsBarcode from 'jsbarcode'

const defaultOptions = {
    imageDomID: 'imageDomID',
    barcodeID: 'barcodeID',
}
describe('Barcode', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <Barcode {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders and calls function', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <Barcode {...options}/>,
    )
    const inst = component.getInstance()
    inst.renderBarcode()
    expect(JsBarcode.mock.calls.length).toBe(1)
  })
})
