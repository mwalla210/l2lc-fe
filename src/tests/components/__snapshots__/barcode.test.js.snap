// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Barcode Renders with snapshot 1`] = `
<img
  alt="image"
  id="imageDomID"
  onLoad={[Function]}
  src="../../style/open-iconic-master/svg/image.svg"
/>
`;
